libmamba/src/solv-cpp/solvable.cpp

```diff
@@ -349,37 +349,32 @@ namespace mamba::solv
         return set_subdir(str.c_str());
     }
 
-    auto ObjSolvableViewConst::dependencies() const -> ObjQueue
+    auto ObjSolvableViewConst::dependencies(DependencyMarker marker) const -> ObjQueue
     {
         auto q = ObjQueue{};
-        ::solvable_lookup_deparray(
-            const_cast<::Solvable*>(raw()),
-            SOLVABLE_REQUIRES,
-            q.raw(),
-            /* marker= */ -1
-        );
+        ::solvable_lookup_deparray(const_cast<::Solvable*>(raw()), SOLVABLE_REQUIRES, q.raw(), marker);
         return q;
     }
 
-    void ObjSolvableView::set_dependencies(const ObjQueue& q) const
+    void ObjSolvableView::set_dependencies(const ObjQueue& q, DependencyMarker marker) const
     {
-        ::solvable_set_deparray(
-            raw(),
-            SOLVABLE_REQUIRES,
-            const_cast<::Queue*>(q.raw()),
-            /* marker= */ 0
-        );
+        ::solvable_set_deparray(raw(), SOLVABLE_REQUIRES, const_cast<::Queue*>(q.raw()), marker);
     }
 
-    void ObjSolvableView::add_dependency(DependencyId dep) const
+    void ObjSolvableView::add_dependency(DependencyId dep, DependencyMarker marker) const
     {
-        raw()->requires = ::repo_addid_dep(raw()->repo, raw()->requires, dep, /* marker= */ 0);
+        raw()->requires = ::repo_addid_dep(raw()->repo, raw()->requires, dep, marker);
     }
 
     auto ObjSolvableViewConst::provides() const -> ObjQueue
     {
         auto q = ObjQueue{};
-        ::solvable_lookup_deparray(const_cast<::Solvable*>(raw()), SOLVABLE_PROVIDES, q.raw(), -1);
+        ::solvable_lookup_deparray(
+            const_cast<::Solvable*>(raw()),
+            SOLVABLE_PROVIDES,
+            q.raw(),
+            /* marker= */ -1
```

Comment:
Do we know why we need this magic number here?

Reply:
This was only reformated by I believe this magic number was already present pre-refactoring.

The idea is that dep array can be split in to with a magic marker. When retrieving the dependencies, `-1` means first part, `1` means second and 0 means everything, including said marker (which is not a valid dependency).
That works if the default maker was used when setting the dependencies...
So yes, a bit of a mess.

See also the test case added in this PR. So far we have no need for markers in `provide` key so we keep sane defaults.